src/transformers/models/mask2former/configuration_mask2former.py

```diff
@@ -0,0 +1,309 @@
+            Number of points used for sampling during loss calculation.
+        oversample_ratio (`float`, *optional*, defaults to 3.0):
+            Oversampling parameter used for calculating no. of sampled points
+        importance_sample_ratio (`float`, *optional*, defaults to 0.75):
+            Ratio of points that are sampled via importance sampling.
+        init_std (`float`, *optional*, defaults to 0.02):
+            The standard deviation of the truncated_normal_initializer for initializing all weight matrices.
+        init_xavier_std (`float``, *optional*, defaults to 1.0):
+            The scaling factor used for the Xavier initialization gain in the HM Attention map module.
+        use_auxiliary_loss (`boolean``, *optional*, defaults to `True`):
+            If `True` [`Mask2FormerForUniversalSegmentationOutput`] will contain the auxiliary losses computed using
+            the logits from each decoder's stage.
+        feature_strides (`List[int]`, *optional*, defaults to [4, 8, 16, 32]):
+            Feature strides corresponding to features generated from backbone network.
+        output_auxiliary_logits (`bool`, *optional*):
+            Should the model output its `auxiliary_logits` or not.
+
+    Examples:
+
+    ```python
+    >>> from transformers import Mask2FormerConfig, Mask2FormerModel
+
+    >>> # Initializing a Mask2Former facebook/mask2former-instance-swin-small-coco configuration
+    >>> configuration = Mask2FormerConfig()
+
+    >>> # Initializing a model (with random weights) from the facebook/mask2former-instance-swin-small-coco style configuration
+    >>> model = Mask2FormerModel(configuration)
+
+    >>> # Accessing the model configuration
+    >>> configuration = model.config
+    ```
+
+    """
+    model_type = "mask2former"
+    attribute_map = {"hidden_size": "mask_feature_size"}
+
+    def __init__(
+        self,
+        backbone_config: Optional[Dict] = None,
+        decoder_config: Optional[Dict] = None,
+        ignore_value: Optional[int] = 255,
+        num_queries: Optional[int] = 150,
+        no_object_weight: Optional[float] = 0.1,
+        class_weight: Optional[float] = 2.0,
+        mask_weight: Optional[float] = 5.0,
+        dice_weight: Optional[float] = 5.0,
+        train_num_points: Optional[int] = 12544,
+        oversample_ratio: Optional[float] = 3.0,
+        importance_sample_ratio: Optional[float] = 0.75,
+        init_std: Optional[float] = 0.02,
+        init_xavier_std: Optional[float] = 1.0,
+        use_auxiliary_loss: Optional[bool] = True,
+        feature_strides: Optional[List[int]] = [4, 8, 16, 32],
+        output_auxiliary_logits: Optional[bool] = None,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+
+        if backbone_config is None:
+            logger.info("`backbone_config` is `None`. Initializing the config with the default `Swin` backbone.")
+            backbone_config = SwinConfig(
```

Comment:
To be consistent with @sgugger's comment in the UperNet PR, we can use `CONFIG_MAPPING["swin"]` here instead of `SwinConfig`, to remove the dependency on Swin in the configuration class of Mask2Former.

Although I'm more in favor actually of instantiating an explicit `SwinConfig`, not sure what you think @alaradirik ?

Reply:
Yeah see [this comment](https://github.com/huggingface/transformers/pull/20648#discussion_r1055135634) by @sgugger for why we may want to use CONFIG_MAPPING. But I'm also in favor of making it more explicit